modules/auxiliary/dos/windows/http/rb_dos_iis_2021_31166.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+###
+#
+# This module performs a DOS attack using a simple HTTP request.
+#
+###
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Dos
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'CVE-2021-31166: HTTP Protocol Stack Remote Code Execution' +
+          ' Vulnerability - Windows IIS DOS BlueScreen',
+        'Description' => 'This module can be used to perform a DOS attack on' +
+          ' IIS server. This module exploit CVE-2021-31166 and causes a Blue' +
+          ' Screen with only one payload.',
+        'License' => MSF_LICENSE,
+        'Author' => ['Maurice LAMBERT <mauricelambert434@gmail.com>'],
+        'Platform' => 'win',
+        'References' => [
+          ['CVE', '2021-31166'],
+          ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2021-31166'],

```

Comment:
Should add another URL reference here for the MSRC case for this and also to https://twitter.com/metr0/status/1392631376592076805 which explains the original bug finders as [@_mxms](https://twitter.com/_mxms) and [@fzzyhd1](https://twitter.com/fzzyhd1).